ingest-app/src/cmr/ingest/data/memory_db.clj

```diff
@@ -55,21 +58,24 @@
 
   (create-and-save-bulk-update-status
     [this provider-id json-body concept-ids]
-    (swap! task-id-atom inc)
-    (let [task-id @task-id-atom
-          task-statuses @task-status-atom
-          collection-statuses @collection-status-atom]
-     (reset! task-status-atom (conj task-statuses
-                                    {:task-id task-id
-                                     :provider-id provider-id
-                                     :request-json-body json-body
-                                     :status "IN_PROGRESS"}))
-     (reset! collection-status-atom (concat collection-statuses
-                                            (map (fn [c]
-                                                  {:task-id task-id
-                                                   :concept-id c
-                                                   :status "PENDING"})
-                                                 concept-ids)))))
+    (locking locking-object
```

Comment:
We should use `swap!` to modify the atoms, then we don't need `locking` and `reset!`.